crates/polars-ops/src/series/ops/is_in.rs

```diff
@@ -556,6 +560,29 @@ fn is_in_struct(ca_in: &StructChunked, other: &Series) -> PolarsResult<BooleanCh
     }
 }
 
+#[cfg(feature = "dtype-categorical")]
```

Comment:
Hi @ritchie46, is this more in line with what you were thinking?

I imagine it can be improved--I'm not sure how to deal with the identical `apply_values_generic` lines. I understand that the hash set captures references that don't live longer than the if/else statement but not entirely sure what to do about it.

Reply:
What I'm trying to resolve is this:

```rust
    // We need only check against the unique categories used by this array.
    let set = if other._can_fast_unique() {
        // Build hash set directly from categories
        let cats = other.get_rev_map().get_categories();
        PlHashSet::from_iter(cats.values_iter())
    } else {
        // Build hash set from unique values
        let cats = other.unique().unwrap();
        PlHashSet::from_iter(cats.iter_str().flatten())
    };
    Ok(ca_in
        .apply_values_generic(|val| set.contains(&val))
        .with_name(ca_in.name())
    )
```
```
569 |     let set = if other._can_fast_unique() {
    |         --- borrow later stored here
...
575 |         let cats = other.unique().unwrap();
    |             ---- binding `cats` declared here
576 |         PlHashSet::from_iter(cats.iter_str().flatten())
    |                              ^^^^ borrowed value does not live long enough
577 |     };
    |     - `cats` dropped here while still borrowed
```
I think I get what's happening but not sure how to fix...any quick tips?

Reply:
I can also do:
 
```python
PlHashSet::from_iter(cats.values_iter().map(|s| s.to_owned())
...
PlHashSet::from_iter(cats.iter_str().flatten().map(|s| s.to_owned())
```